### PR TITLE
feat(robot): add robot team membership

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
uses https://github.com/upbound/up-sdk-go/pull/93
implements an option to assign an robot to a team 

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
up robot team list chris -a upboundcare
TEAMS
test678
```

```
up robot team join test123 chris -a upboundcare 
Adding robot "chris" to team "test123"
```

```
up robot team list chris -a upboundcare
TEAMS
test678
test123
```

```
up robot team leave test123 chris -a upboundcare
Are you sure you want to remove this robot from the team? This cannot be undone [y/n]: y
Removing robot "chris" from team "test123"
Removed robot "chris" from team "test123"
```

### CODE REVIEW

File: cmd/up/robot/membership/create.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	rs, err := oc.ListRobots(ctx, a.Organization.ID)
+	if err != nil {
+		return err
+	}
+	if len(rs) == 0 {
+		return errors.Errorf(errFindRobotFmt, c.RobotName, upCtx.Account)
+	}
+
+	// Ensure exactly one robot with the specified name exists
+	var robotID uuid.UUID
```
Comment (RedbackThomson) [inline cmd/up/robot/membership/create.go:56]: Is this necessary?

File: cmd/up/robot/membership/delete.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	return nil
+}
+
+// AfterApply accepts user input by default to confirm the delete operation.
+func (c *deleteCmd) AfterApply(p pterm.TextPrinter, upCtx *upbound.Context) error {
+	if c.Force {
+		return nil
+	}
+
+	confirm, err := c.prompter.Prompt("Are you sure you want to delete this robot team membership? This cannot be undone [y/n]", false)
```
Comment (RedbackThomson) [inline cmd/up/robot/membership/delete.go:45]: I'm not sure how I feel about calling this a `robot team membership`. Users aren't explicitly manipulating "membership" objects, they're adding and removing members from teams. I think it's more intuitive to just say:
```
Are you sure you want to remove this robot from the team?
```

File: cmd/up/robot/membership/delete.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	if err != nil {
+		return err
+	}
+
+	if input.InputYes(confirm) {
+		p.Printfln("Deleting robot team membership %s/%s/%s.", upCtx.Account, c.RobotName, c.TeamName)
```
Comment (RedbackThomson) [inline cmd/up/robot/membership/delete.go:51]:
```
Removing robot %q from team %q
```
?

File: cmd/up/robot/membership/get.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	"github.com/upbound/up-sdk-go/service/accounts"
+	"github.com/upbound/up-sdk-go/service/organizations"
+	"github.com/upbound/up-sdk-go/service/robots"
+	"github.com/upbound/up-sdk-go/service/teams"
+
+	"github.com/upbound/up/internal/upbound"
+	"github.com/upbound/up/internal/upterm"
+)
+
+var fieldNames = []string{"TEAM MEMBERSHIP"}
```
Comment (RedbackThomson) [inline cmd/up/robot/membership/get.go:34]:
```suggestion
var fieldNames = []string{"TEAMS"}
```

File: cmd/up/robot/membership/membership.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	kongCtx.Bind(robots.NewClient(cfg))
+	kongCtx.Bind(teams.NewClient(cfg))
+	return nil
+}
+
+// Cmd contains commands for managing robot team memberships.
+type Cmd struct {
+	Create createCmd `cmd:"" help:"Create a robot team membership."`
+	Delete deleteCmd `cmd:"" help:"Delete a robot team membership."`
+	Get    getCmd    `cmd:"" help:"Get all robot team memberships."`
```
Comment (RedbackThomson) [inline cmd/up/robot/membership/membership.go:49]: Perhaps `join`, `leave` and `get` are more appropriate?

File: cmd/up/robot/robot.go [hunk 6]
```diff
@@ -95,11 +96,12 @@ func PredictRobots() complete.Predictor {
...
-	Delete deleteCmd `cmd:"" help:"Delete a robot."`
-	List   listCmd   `cmd:"" help:"List robots for the account."`
-	Get    getCmd    `cmd:"" help:"Get a robot for the account."`
-	Token  token.Cmd `cmd:"" help:"Interact with robot tokens."`
+	Create     createCmd      `cmd:"" help:"Create a robot."`
+	Delete     deleteCmd      `cmd:"" help:"Delete a robot."`
+	List       listCmd        `cmd:"" help:"List robots for the account."`
+	Get        getCmd         `cmd:"" help:"Get a robot for the account."`
+	Token      token.Cmd      `cmd:"" help:"Interact with robot tokens."`
+	Membership membership.Cmd `cmd:"" help:"Interact with robot team membership."`
```
Comment (RedbackThomson) [inline cmd/up/robot/robot.go:104]: Instead of membership, team associations?

Comment (adamwg) [reply cmd/up/robot/robot.go:104]: I would lean toward just calling the command `team` - `up robot team [list|join|leave]`.

File: cmd/up/robot/membership/create.go (hunk 1 above)
Comment (haarchri) [reply cmd/up/robot/membership/create.go]: like in our other commands yes

**Review by RedbackThomson — APPROVED** [review]

**Pull request merged**